app/assets/javascripts/components/common/ArticleViewer/containers/ArticleViewer.jsx

```diff
@@ -63,9 +63,21 @@ export class ArticleViewer extends React.Component {
   // first in that case. In that case, componentDidUpdate fetches the
   // user ids as soon as usernames are avaialable.
   componentDidUpdate(prevProps, prevState) {
-    if (!prevProps.users && this.props.users) {
-      if (!prevState.userIdsFetched) {
-        this.fetchUserIds(this.props.users);
+    // checks for the availability of 'assignedUsers' and 'users' props.
+    // This condition will only pass when articleViewer is accessed from the Students/Editors tab
+    if (this.props.assignedUsers && this.props.users) {
+    // checks if usersIds required for coloration have already been fetched or not.
+    // if not, userIds are fetched by passing in an array that contains both 'assignedUsers'
+    // and 'users' as an argument to the fetchUserIds function.
+      if (!this.state.userIdsFetched) {
```

Comment:
Why does this switch on `state.userIdsFetched` but the other branch switches on `prevState.userIdsFetched`?

It seems like any time `prevState.userIdsFetched` is true, `state.userIdsFetched` will also be true, since it just goes from false to true, not the other way around.

If that's the case, I suggest using a guard clause at the start so that less nesting and repetition is required.